executor/executor_test.go

```diff
@@ -6800,6 +6800,11 @@ func (s *testSuite1) TestUpdateGivenPartitionSet(c *C) {
 
 	tk.MustExec("update t2 partition(p0) set a = 3 where a = 2")
 	tk.MustExec("update t2 partition(p0, p3) set a = 33 where a = 1")
+
+	tk.MustExec("create table t4(a int primary key, b int) partition by hash(a) partitions 2")
+	tk.MustExec("insert into t4(a, b) values(1, 1),(2, 2),(3, 3);")
+	err = tk.ExecToErr("update t4 partition(p0) set a = 6 where a = 2")
```

Comment:
The test case is not correct, both a = 6 and a = 2 are in the same partition p0.
You get to check the original test case in the issue. @TszKitLo40